File: models/opd/OpdToken.js
import mongoose from "mongoose";
const OpdTokenSchema = new mongoose.Schema({
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },
  roomId: { type: mongoose.Schema.Types.ObjectId, ref: "Room", required: true },
  doctorId: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  patientId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientRegistration", required: true },
  visitId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientVisit", required: true },
  tokenDate: { type: Date, required: true },
  tokenNumber: { type: Number, required: true },
  priority: { type: String, enum: ["Normal", "Urgent"], default: "Normal" },
  status: { type: String, enum: ["Waiting", "Called", "In-Consultation", "Completed", "No-Show", "Cancelled"], default: "Waiting" },
  calledAt: { type: Date },
  completedAt: { type: Date }
}, { timestamps: true });
export default mongoose.model("OpdToken", OpdTokenSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/OpdToken.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "opd_tokens")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OpdToken extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    private Room room;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id")
    private User doctor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private PatientRegistration patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "visit_id", nullable = false)
    private PatientVisit visit;

    @Column(name = "token_date", nullable = false)
    private LocalDateTime tokenDate;

    @Column(name = "token_number", nullable = false)
    private Integer tokenNumber;

    @Enumerated(EnumType.STRING)
    private Priority priority = Priority.Normal;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Waiting;

    @Column(name = "called_at")
    private LocalDateTime calledAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    public enum Priority {
        Normal, Urgent
    }

    public enum Status {
        Waiting, Called, In_Consultation, Completed, No_Show, Cancelled
    }
}
